lightning/src/ln/onchain_utils.rs

```diff
@@ -0,0 +1,892 @@
+				PackageTemplate::HolderHTLCTx {
+					input: (outpoint, htlc_outp)
+				}
+			},
+			3 => {
+				let outpoint = Readable::read(reader)?;
+				let funding_outp = Readable::read(reader)?;
+				PackageTemplate::HolderCommitmentTx {
+					input: (outpoint, funding_outp)
+				}
+			},
+			_ => return Err(DecodeError::InvalidValue),
+		};
+		Ok(package)
+	}
+}
+
+/// BumpStrategy is a basic enum to encode a fee-committing strategy. We
+/// may extend it in the future with other stategies like BYOF-input.
+#[derive(PartialEq, Clone)]
+pub(crate) enum BumpStrategy {
+	RBF,
+	CPFP
+}
+
+impl Writeable for BumpStrategy {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), ::std::io::Error> {
+		match self {
+			BumpStrategy::RBF => {
+				writer.write_all(&[0; 1])?;
+			},
+			BumpStrategy::CPFP => {
+				writer.write_all(&[1; 1])?;
+			}
+		}
+		Ok(())
+	}
+}
+
+impl Readable for BumpStrategy {
+	fn read<R: ::std::io::Read>(reader: &mut R) -> Result<Self, DecodeError> {
+		let bump_strategy = match <u8 as Readable>::read(reader)? {
+			0 => {
+				BumpStrategy::RBF
+			},
+			1 => {
+				BumpStrategy::CPFP
+			},
+			_ => return Err(DecodeError::InvalidValue),
+		};
+		Ok(bump_strategy)
+	}
+}
+
+/// A structure to describe a claim content and its metadatas which is generated
+/// by ChannelMonitor and used by OnchainTxHandler to generate feerate-competive
+/// transactions.
+///
+/// Metadata are related to multiple fields playing a role in packet lifetime.
+/// Once issued, it may be aggregated with other requests if it's judged safe
+/// and feerate opportunistic.
+/// Current LN fees model, pre-committed fees with update_fee adjustement, means
+/// that counter-signed transactions must be CPFP to be dynamically confirmed as a
+/// bumping strategy. If transactions aren't lockdown (i.e justice transactions) we
+/// may RBF them.
+/// Feerate previous will serve as a feerate floor between different bumping attempts.
+/// Height timer clocks these different bumping attempts.
+/// Absolute timelock defines the block barrier at which claiming isn't exclusive
+/// to us anymore and thus we MUST have get it solved before.
+/// Height original serves as a packet timestamps to prune out claim in case of reorg.
+/// Content embeds transactions elements to generate transaction. See PackageTemplate.
+#[derive(PartialEq, Clone)]
+pub struct OnchainRequest {
+	// Timeout tx must have nLocktime set which means aggregating multiple
+	// ones must take the higher nLocktime among them to satisfy all of them.
+	// Sadly it has few pitfalls, a) it takes longuer to get fund back b) CLTV_DELTA
+	// of a sooner-HTLC could be swallowed by the highest nLocktime of the HTLC set.
+	// Do simplify we mark them as non-aggregable.
+	pub(crate) aggregation: bool,
+	// Content may lockdown with counter-signature of our counterparty
+	// or fully-malleable by our own. Depending on this bumping strategy
+	// must be adapted.
+	pub(crate) bump_strategy: BumpStrategy,
```

Comment:
Isn't this determined by the actual output itself? Shouldn't this be in the PackageTemplate?

Reply:
Added a is_malleable() to PackageTemplate, we should get the same effect than BumpStrategy. Yes for now bump_strategy == malleability, but it might differ in the future with watchtowers/channel monitors replicas and where the keys are laying.